resonate/scheduler.py

```diff
@@ -712,6 +712,15 @@ def _eval(self, node: Node[State]) -> list[Function | Delayed[Function | Retry]
                         node.transition(Enabled(Completed(f.map(result=result))))
                         self._unblock(suspends, result)
                         return []
+                    case Ko(), delay, False if type(result.value) in opts.non_retriable_errors:
```

Comment:
```suggestion
                    case Ko(v), delay, False if type(v) in opts.non_retriable_errors:
```

Reply:
Also mind switching the order of the durable case? The order right now is false then true, let's do true then false